libs/widget-react/src/lib/CowSwapWidget.tsx

```diff
@@ -2,26 +2,21 @@ import { useCallback, useEffect, useRef, useState } from 'react'
 
 import {
   createCowSwapWidget,
-  CowSwapWidgetParams,
   EthereumProvider,
   CowSwapWidgetHandler,
+  CowSwapWidgetProps,
+  CowSwapWidgetParams,
 } from '@cowprotocol/widget-lib'
 import type { CowEventListeners } from '@cowprotocol/events'
 
-export interface CowSwapWidgetProps {
-  params: Omit<CowSwapWidgetParams, 'provider'>
-  provider?: EthereumProvider
-  listeners?: CowEventListeners
-}
-
 import { Command } from '@cowprotocol/types'
 
-export function CowSwapWidget({ params, provider, listeners }: CowSwapWidgetProps) {
+export function CowSwapWidget(props: CowSwapWidgetProps) {
+  const { params, provider, listeners } = props
   const [error, setError] = useState<{ error: Error; message: string } | null>(null)
-  const paramsRef = useRef<Omit<CowSwapWidgetParams, 'provider'> | null>(null)
-  const providerRef = useRef<EthereumProvider | null>(provider ?? null)
+  const paramsRef = useRef<CowSwapWidgetParams | null>(null)
+  const providerRef = useRef<EthereumProvider | undefined>(provider)
```

Comment:
just nit: code got simpler as I was converting from null to undefined and the other way around